conf/web2py

```diff
@@ -17,20 +18,28 @@ git clone --recursive --branch $VER https://github.com/web2py/web2py
 
 # Install GitPython in virtualenv to workaround turnkey-pylib
 # git.py conflict
-virtualenv $W2PROOT/venv
-$W2PROOT/venv/bin/pip install GitPython
+virtualenv -p python2 $W2PROOT/venv2
+$W2PROOT/venv2/bin/pip install GitPython
+
+# Also install py3 venv - to add support for python3
+virtualenv -p python3 $W2PROOT/venv3
+
+# Use python2 by default for now
+ln -s $W2PROOT/venv2 $W2PROOT/venv
 
 chown -R www-data:www-data $W2PROOT
 
 # copy wsgihandler to web2py root
-cp $W2PROOT/handlers/wsgihandler.py $W2PROOT/wsgihandler.py
+cp $W2PROOT/handlers/wsgihandler.py $WSGI
 
-# Code for entering virtualenv after wsgihandler is started
-VIRTUALENV_ACTIVATE_PATH='/var/www/web2py/venv/bin/activate_this.py'
-VIRTUALENV_ACTIVATE="activate_this = '$VIRTUALENV_ACTIVATE_PATH'\nexecfile(activate_this, dict(__file__=activate_this))"
-
-# Insert code just below initial imports
-sed -i "s|import os|import os\n\n$VIRTUALENV_ACTIVATE|" $W2PROOT/wsgihandler.py
+# Add code for entering virtualenv after wsgihandler is started
+#
+# Insert placeholders
+sed -i 's|import os|import os\n\nLINE_1\nLINE_2\nLINE_3|' $WSGI
+# Insert proper lines
+sed -i "s|LINE_1|activate_this = '$W2PROOT/venv/bin/activate_this.py'|" $WSGI
+sed -i "s|LINE_2|with open(activate_this, 'rb') as fob:|" $WSGI
+sed -i "s|LINE_3|    exec(compile(fob.read(), activate_this, 'exec'), dict(__file__=activate_this))|" $WSGI
```

Comment:
Why not use sed's `a` (append) for adding these lines rather than substitution?

Reply:
Yeah, you're probably right. I did initially try to do it via an alternate way but I couldn't get it to work and IMO doing it the way I did makes it relatively readable. Regardless, I'm going to leave it as is for now...